File: algorithms/2_medium/C/03876_construct_uniform_parity_array_ii/solution.cpp
class Solution {
public:
    static bool uniformArray(const vector<int>& nums1) {
        assert(!nums1.empty());

        int min = numeric_limits<int>::max();
        bool has_odd = false;

        for (const int num : nums1) {
            min = std::min(min, num);
            if ((num & 1) == 1) {
                has_odd = true;
            }
        }

        if ((min & 1) == 0 && has_odd) {
            return false;
        }

        return true;
    }
};

File: algorithms/2_medium/C/03876_construct_uniform_parity_array_ii/solution.rs
impl Solution {
    pub fn uniform_array(nums1: Vec<i32>) -> bool {
        let (mut min, mut has_odd): (Option<i32>, bool) = (None, false);
        for num in nums1 {
            min = Some(min.unwrap_or(num).min(num));
            if (num & 1) == 1 {
                has_odd = true; 
            }
        }

        match min {
            Some(min_val) => {
                if (min_val & 1) == 0 && has_odd {
                    return false;
                }

                true
            },
            None => unreachable!(),
        }
    }
}
